submit.go

```diff
@@ -232,7 +230,59 @@ type whoamiResponse struct {
 	} `json:"matrix"`
 }
 
-func (s *submitServer) verifyAccessToken(ctx context.Context, auth, userID string) (*whoamiResponse, error) {
+type imaWhoamiResponse struct {
+	IMAUserToken string `json:"ima_user_token"`
+	AnalyticsID  string `json:"analytics_id"`
+	Email        string `json:"email"`
+	Subscription struct {
+		ExpiresAt string `json:"expires_at"`
+		Active    bool   `json:"active"`
+	} `json:"subscription"`
+}
+
+func (s *submitServer) verifyIMAToken(ctx context.Context, auth, userID string) (*imaWhoamiResponse, error) {
+	if len(auth) == 0 {
+		return nil, fmt.Errorf("missing authorization header")
+	} else if !strings.HasPrefix(auth, "Bearer ") {
+		return nil, fmt.Errorf("invalid authorization header")
+	}
+
+	// The user ID in this case should be an email
+	atIndex := strings.IndexRune(userID, '@')
+	if atIndex <= 0 {
+		return nil, fmt.Errorf("invalid user ID")
+	}
+
+	// All of iMessage on Android is on beeper.com
+	apiServerURL, ok := s.cfg.APIServerURLs["beeper.com"]
```

Comment:
I don't think that's strictly true, this should probably use the current env api server (then dev/staging booper can be pointed at dev/staging rageshake)

Reply:
Oh wait, we might not be telling rageshake which env it is in 🤔 I guess hardcoding beeper.com for now is easier

Reply:
Not sure if there's a switcher, but booper used to be pointed at staging until a week ago or something